src/arithmetic/aggregate_funcs/agg_funcs.c

```diff
@@ -73,32 +73,58 @@ AggregateResult AGG_SUM(SIValue *argv, int argc) {
 typedef struct {
 	size_t count;
 	double total;
+	bool overflow;
 } _agg_AvgCtx;
 
 AggregateResult AGG_AVG(SIValue *argv, int argc) {
 	AggregateCtx *ctx = argv[1].ptrval;
-	// On the first invocation, initialize the context.
+	// on the first invocation, initialize the context
 	if(ctx->private_ctx == NULL) ctx->private_ctx = rm_calloc(1, sizeof(_agg_AvgCtx));
 
-	SIValue v = argv[0];
-	if(SI_TYPE(v) == T_NULL) return AGGREGATE_OK;
+	SIValue si_val = argv[0];
+	if(SI_TYPE(si_val) == T_NULL) return AGGREGATE_OK;
+	long double v = SI_GET_NUMERIC(si_val);
 
 	_agg_AvgCtx *avg_ctx = ctx->private_ctx;
 
-	// If we're uniquing inputs, return early if this value has already been seen.
-	if(ctx->hashSet && Set_Add(ctx->hashSet, v) == false) return AGGREGATE_OK;
-
-	avg_ctx->count ++;
-	avg_ctx->total += SI_GET_NUMERIC(v);
+	// if we're uniquing inputs, return early if this value has already been seen
+	if(ctx->hashSet && Set_Add(ctx->hashSet, si_val) == false) return AGGREGATE_OK;
+
+	avg_ctx->count ++; // increment the count
+
+	// if we've already overflowed or adding the current value
+	// will cause us to overflow, use the incremental averaging algorithm
+	if(unlikely(avg_ctx->overflow || // already reached overflow
+				(signbit(avg_ctx->total) == signbit(v) && // each value has same sign
+				 (fabs(avg_ctx->total) > (DBL_MAX - fabs(v)))))) { // about to overflow
```

Comment:
we're getting quite a lot of `Codecov` complaints here, let make sure each line here is covered.

Reply:
The test was broken before, fixed now!